### DISCUSSION TITLE
"Cannot find package '@graphql-mesh/transport-rest'" when running gateway package container

### DISCUSSION THREAD

**Discussion by blakel6819** [graphql-hive/gateway#235] [Q&A] [answered]
### Description
Getting the following error running any of the 1.5.x  gateway package containers when trying to load transport "rest":
- `Error [ERR_MODULE_NOT_FOUND]: Cannot find package '@graphql-mesh/transport-rest' imported from /node_modules/.chunk/unifiedGraphManager-DAEQW3BS.mjs`

Can the `@graphql-mesh/transport-rest package` be included in gateway package container?
Or is there an alternative way to get this working?

### Additional Context
I am defining a subgraph from an OpenAPI schema using `@omnigraph/openapi` package and [following the GraphQL-Mesh docs](https://the-guild.dev/graphql/mesh/v1/source-handlers/openapi).

The package is installed and is available in the  node_modules directory, but it is not part of the container image.

Everything works as expected when running locally, but fails when running in Docker (locally and deployed).

If I try to import the package directly from the node_modules directory, I then encounter the following duplicate graphql modules error:
- Duplicate "graphql" modules cannot be used at the same time since different versions may have different capabilities and behavior. The data from one version used in the function from another could produce confusing and spurious results.
- https://github.com/ardatan/graphql-js/blob/master/src/jsutils/instanceOf.js#L27
- I have tried using resolutions in the pnpm package.json file as well

### Environment
- @graphql-hive/gateway: ^1.5.x
- @graphql-mesh/compose-cli: ^1.2.x
- @omnigraph/openapi: ^0.108.x

**Comment by enisdenjo** [comment]
Not all transports are included in the docker image. You have to install it yourself. Please check [the "Extend Docker Image" documentation](https://the-guild.dev/graphql/hive/docs/gateway/deployment/docker#extend-docker-image) on the how-to.

Your Dockerfile should roughly be:
```dockerfile
FROM ghcr.io/graphql-hive/gateway
 
RUN npm i @graphql-mesh/transport-rest
```

**Reply by blakel6819** [reply]
Thanks for your reply @enisdenjo.

I have tried the suggestion, but still getting the following duplicate graphql modules error:
- Duplicate "graphql" modules cannot be used at the same time since different versions may have different capabilities and behavior. The data from one version used in the function from another could produce confusing and spurious results.
- https://github.com/ardatan/graphql-js/blob/master/src/jsutils/instanceOf.js#L27
- I have tried using resolutions in the pnpm package.json file as well

The repo I am working in is a monorepo using turborepo, could that be causing issues extending the docker image?

**Reply by ardatan** [reply]
But this command has nothing to do with your local setup;
```Dockefile
RUN npm i @graphql-mesh/transport-rest
```
It installs the dependency inside the container.

Could you help us to reproduce your issue within a repo like this?
https://github.com/ardatan/docker-mesh-openapi-v1

**Reply by simonleebaker** [reply]
Hi @enisdenjo / @ardatan , I came across this as I am experiencing the same issue with not being able to locate the '@graphql-mesh/transport-rest' package. I've cloned the repo posted by @ardatan and having made no changes to that, the issue is present when I run that too, so would appreciate any advice. Thanks in advance!

**Reply by simonleebaker** [reply]
Just to add, the transport-rest has been installed, but to the gateway directory. It is not present in the top level node_modules directory.

<img width="564" alt="total 36" src="https://github.com/user-attachments/assets/a8784fc4-3fb3-4c2e-8c96-2d113401d5e7" />

<img width="564" alt="$ cd @graphql-mesh" src="https://github.com/user-attachments/assets/04c3f0d8-1a65-4f7c-89b5-f4df31615f83" />

**Reply by ardatan** [reply]
Could you help us to reproduce your issue within a repo like this?
https://github.com/ardatan/docker-mesh-openapi-v1 @simonleebaker

**Reply by simonleebaker** [reply]
Thanks @ardatan - that is exactly the repo I have used to reproduce the issue.

**Reply by ardatan** [reply]
I see but we still don't know about your setup, configuration and more. We'd love to reproduce it exactly the same way you did. So in order to help you better, could you **_please_** provide a reproduction within a repo like I shared above?

**Reply by simonleebaker** [reply]
I've created a repo that shows the problem [here](https://github.com/simonleebaker/openai-gateway/) @ardatan and included the steps to reproduce in the readme. Hope this is helpful.

**Reply by ardatan** [reply]
Thank you @simonleebaker ! In this way, we can eliminate any extra trial and error steps!

**Reply by simonleebaker** [reply]
@ardatan I think I've found what is causing the issue - without the 'transports' set in the gateway configuration, it appears that the gateway cannot find the package, regardless of it being installed via npm or not.

<img width="571" alt="transports" src="https://github.com/user-attachments/assets/971cdd8b-5fe1-49ea-a121-60d201345451" />

When I try it with the transports populated as above, it works as expected :-)

**Reply by enisdenjo** [reply]
Glad you got it working! That's certainly a way of doing it. But what you've been experiencing is definitely a bug and we're on fixing it.

**Comment by enisdenjo** [comment, marked as answer]
This has been fixed in #865 and released in v1.13.0 of the Hive Gateway Docker image, i.e. starting from `ghcr.io/graphql-hive/gateway:1.13.0`.
